pkg/ako/values.go

```diff
@@ -297,21 +307,35 @@ type L7Settings struct {
 	NoPGForSNI           bool   `yaml:"no_pg_for_SNI"`
 }
 
+type ServiceType string
+
+const (
+	NodePort      ServiceType = "NodePort"
+	ClusterIP     ServiceType = "ClusterIP"
+	NodePortLocal ServiceType = "NodePortLocal"
+)
+
 // DefaultL7Settings returns the default L7Settings
-func DefaultL7Settings() *L7Settings {
+func DefaultL7Settings(cni CNI) *L7Settings {
+	// If antrea is the cluster cni implementation,
+	// the default service will be set to NPL
+	serviceType := NodePort
+	if cni == Antrea {
+		serviceType = NodePortLocal
+	}
 	return &L7Settings{
-		DefaultIngController: false,
-		ServiceType:          "NodePort",
-		ShardVSSize:          "SMALL",
-		// L7ShardingScheme: don't set, use default value in AKO
-		// PassthroughShardSize: don't set, use default value in AKO
+		ServiceType: string(serviceType),
```

Comment:
nit: No need to convert as `serviceType` is of type string

Reply:
it looks like it will complain type error if not convert it to string